api/src/main/java/org/pmiops/workbench/reporting/ReportingServiceImpl.java

```diff
@@ -1,24 +1,52 @@
 package org.pmiops.workbench.reporting;
 
+import javax.inject.Provider;
+import org.pmiops.workbench.config.WorkbenchConfig;
 import org.pmiops.workbench.model.ReportingSnapshot;
+import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Service;
 
 @Service
 public class ReportingServiceImpl implements ReportingService {
 
-  private final ReportingUploadService reportingUploadService;
+  private static enum UploadMethod {
+    DML,
+    STREAMING;
+  }
+
   private final ReportingSnapshotService reportingSnapshotService;
+  private final ReportingUploadService reportingUploadServiceDmlImpl;
+  private final ReportingUploadService reportingUploadServiceStreamingImpl;
+  private final Provider<WorkbenchConfig> workbenchConfigProvider;
 
   public ReportingServiceImpl(
-      ReportingUploadService reportingUploadService,
-      ReportingSnapshotService reportingSnapshotService) {
-    this.reportingUploadService = reportingUploadService;
+      @Qualifier("REPORTING_UPLOAD_SERVICE_DML_IMPL")
```

Comment:
Note  I'm testing two different implementations of upload here. All this fanciness is really only necessary because my local DB doesn't have nearly as many rows as the Test DB, but I don't want to get into a one-off deployments. We can also compare test against Perf, for example.